### PR TITLE
neovim 0.8.0

### PR DESCRIPTION
**Pull request by n33pm**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by codykrieger** [comment]
With these changes, neovim 0.8 builds for me (macOS 12.6/21G115 on Apple Silicon):

```diff
diff --git i/Formula/neovim.rb w/Formula/neovim.rb
index 62b7fbdd8bd..edd66fb5272 100644
--- i/Formula/neovim.rb
+++ w/Formula/neovim.rb
@@ -6,8 +6,8 @@ class Neovim < Formula
 
   # Remove `stable` block when `gperf` is no longer needed.
   stable do
-    url "https://github.com/neovim/neovim/archive/v0.7.2.tar.gz"
-    sha256 "ccab8ca02a0c292de9ea14b39f84f90b635a69282de38a6b4ccc8565bc65d096"
+    url "https://github.com/neovim/neovim/archive/v0.8.0.tar.gz"
+    sha256 "505e3dfb71e2f73495c737c034a416911c260c0ba9fd2092c6be296655be4d18"
 
     # Libtool is needed to build `libvterm`.
     # Remove this dependency when we use the formula.
@@ -15,13 +15,6 @@ class Neovim < Formula
     # GPerf was removed in https://github.com/neovim/neovim/pull/18544.
     # Remove dependency when relevant commits are in a stable release.
     uses_from_macos "gperf" => :build
-
-    # TODO: Use `libvterm` formula when the following is released:
-    # https://github.com/neovim/neovim/pull/17329
-    resource "libvterm" do
-      url "https://www.leonerd.org.uk/code/libvterm/libvterm-0.1.4.tar.gz"
-      sha256 "bc70349e95559c667672fc8c55b9527d9db9ada0fb80a3beda533418d782d3dd"
-    end
   end
 
   livecheck do
@@ -39,12 +32,11 @@ class Neovim < Formula
     sha256 x86_64_linux:   "be762f679f83c41fb982d32f067229ca3480f991fdcbc7b15147fdef932312a1"
   end
 
-  # Remove `head` block when `stable` depends on `libvterm`.
   head do
     url "https://github.com/neovim/neovim.git", branch: "master"
-    depends_on "libvterm"
   end
 
+  depends_on "libvterm"
   depends_on "cmake" => :build
   depends_on "luarocks" => :build
   depends_on "pkg-config" => :build
@@ -103,14 +95,6 @@ class Neovim < Formula
           end
         end
       end
-
-      if build.stable?
-        # Build libvterm. Remove when we use the formula.
-        cd "libvterm" do
-          system "make", "install", "PREFIX=#{buildpath}/deps-build", "LDFLAGS=-static #{ENV.ldflags}"
-          ENV.prepend_path "PKG_CONFIG_PATH", buildpath/"deps-build/lib/pkgconfig"
-        end
-      end
     end
 
     # Point system locations inside `HOMEBREW_PREFIX`.
@@ -127,12 +111,6 @@ class Neovim < Formula
                     "-DLIBUV_LIBRARY=#{Formula["libuv"].opt_lib/shared_library("libuv")}",
                     *std_cmake_args
 
-    # Patch out references to Homebrew shims
-    # TODO: Remove conditional when the following PR is included in a release.
-    # https://github.com/neovim/neovim/pull/19120
-    config_dir_prefix = build.head? ? "cmake." : ""
-    inreplace "build/#{config_dir_prefix}config/auto/versiondef.h", Superenv.shims_path/ENV.cc, ENV.cc
-
     system "cmake", "--build", "build"
     system "cmake", "--install", "build"
   end
```

**Comment by codykrieger** [comment]
9ffc1dbf795e added the `config_dir_prefix` conditional at the bottom, so at minimum it seems like that needs to be removed. I'm unsure if the `inreplace` should be reverted to its previous state, which was added in 19fc432f14170c9f54f186e58b65c51c96acc5f9:

```ruby
inreplace "build/config/auto/versiondef.h", Superenv.shims_path/ENV.cc, ENV.cc
```

Seems to run fine without it thus far.

**Comment by codykrieger** [comment]
@carlocab ^ thoughts on the above?

File: Formula/neovim.rb [hunk 1]
```diff
@@ -2,26 +2,11 @@ class Neovim < Formula
...
-    uses_from_macos "gperf" => :build
-
-    # TODO: Use `libvterm` formula when the following is released:
-    # https://github.com/neovim/neovim/pull/17329
-    resource "libvterm" do
-      url "https://www.leonerd.org.uk/code/libvterm/libvterm-0.1.4.tar.gz"
-      sha256 "bc70349e95559c667672fc8c55b9527d9db9ada0fb80a3beda533418d782d3dd"
-    end
+    url "https://github.com/neovim/neovim/archive/v0.8.0.tar.gz"
+    sha256 "505e3dfb71e2f73495c737c034a416911c260c0ba9fd2092c6be296655be4d18"
```
Comment (nerdrew) [inline Formula/neovim.rb:9]: The comment above says to remove the `stable` block when `gperf` isn't needed... not sure what that means, but it looks like it isn't needed...

File: Formula/neovim.rb [hunk 2]
```diff
@@ -39,10 +24,8 @@ class Neovim < Formula
     sha256 x86_64_linux:   "be762f679f83c41fb982d32f067229ca3480f991fdcbc7b15147fdef932312a1"
   end
 
-  # Remove `head` block when `stable` depends on `libvterm`.
   head do
     url "https://github.com/neovim/neovim.git", branch: "master"
-    depends_on "libvterm"
   end
```
Comment (dsully) [inline Formula/neovim.rb:29]: I think this entire block can be removed.

**Comment by nerdrew** [comment]
Looks good to me, but I don't have the "approve" permission...

**Comment by nerdrew** [comment]
This branch works on my machine, also an M1. `brew upgrade -s neovim`...

**Comment by carlocab** [comment]
Nice work, @n33pm, @codykrieger, @nerdrew! These changes are all spot-on.

Regarding the `inreplace` for `Superenv`, what does `neovim --version` return upon building from this branch?

One other thing we want to do is build the tree-sitter parsers and install them into `lib/nvim/parser`. Will add that change shortly.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**